src/isomorphic/modern/class/ReactBaseClasses.js

```diff
@@ -144,7 +144,27 @@ ReactPureComponent.prototype.constructor = ReactPureComponent;
 Object.assign(ReactPureComponent.prototype, ReactComponent.prototype);
 ReactPureComponent.prototype.isPureReactComponent = true;
 
+function ReactAsyncComponent(props, context, updater) {
+  // Duplicated from ReactComponent.
+  this.props = props;
+  this.context = context;
+  this.refs = emptyObject;
+  // We initialize the default updater but the real one gets injected by the
+  // renderer.
+  this.updater = updater || ReactNoopUpdateQueue;
+}
+
+ReactAsyncComponent.prototype = new ComponentDummy();
+ReactAsyncComponent.prototype.constructor = ReactAsyncComponent;
+// Avoid an extra prototype jump for these methods.
+Object.assign(ReactAsyncComponent.prototype, ReactComponent.prototype);
+ReactAsyncComponent.prototype.unstable_isAsyncReactComponent = true;
```

Comment:
Interesting. It might not be noticeable for PureComponent because nobody relies on shouldComponentUpdate for behavior.

However, worst case, I guess this just triggers sync mode.